### PR TITLE
project: ensure yaml load returns a dictionary

### PR DESCRIPTION
**Pull request by cmatsuoka**
PyYAML yaml loading and parsing can return None if given an empty file
or empty string. When validating its yaml file, Snapcraft expects the
result to be subscriptable, so return an empty dictionary instead of None.

Fixes SNAPCRAFT-JQ

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/project/_project_info.py [hunk 1]
```diff
@@ -86,4 +86,7 @@ def _load_yaml(*, yaml_file_path: str) -> OrderedDict:
     except yaml.YAMLError as e:
         raise errors.YamlValidationError(str(e), yaml_file_path) from e
 
+    if yaml_contents is None:
```
Comment (sergiusens) [inline snapcraft/project/_project_info.py]: Can you add a unit test for this, one that of an empty file and validate that the instantiation of a ProjectInfo returns the correct YamlValidationError?

Comment (cmatsuoka) [reply snapcraft/project/_project_info.py]: Unit test added.
